README.md

```diff
@@ -1,13 +1,203 @@
+others - please see: [semver.org](https://semver.org/) for the complete
+reference.
+
+### Formatting Examples
+Examples of what can be seen in practical use:
+- `0.0.0`: First official release — stable interface guaranteed moving forward.
+- `3.0.0-alpha.1`: This would refer to unstable alpha version of v3.0.0. ".1"
+  usually means first alpha of what will become version 3.0.0.
+- `2.1.0-beta+exp.sha.5114f85`: This would denotes a beta build, and gives build
+  metadata, likely showing a specific experiment/commit.
+
+## String Parsing
+One way to create a `SemanticVersion` object is to have the library parse a
+string.  Parsing operates as shown in the table below:
+| Example | Compliant? | Explanation | Parsing Result |
+| - | - | - | - |
+| `1.2.3`| Compliant | As per definition | :green_circle: Parses as is. \\n If these values are in separate variables, it is cheaper to [create directly](#creating-the-object-directly). |
+| `1.2` | Not strictly compliant | Missed `patch` | :yellow_circle: Fails parsing but can be parsed with a switch. |
+| `1` | Not strictly compliant | Misses `minor` and `patch` | :yellow_circle: Fails parsing, but can be parsed with a switch. |
+| `v1.2.3` | Not strictly compliant | Has a leading `v`. Acceptable in documentation. |  :yelow_circle: Fails, but can be parsed using a switch that drops the leading `v`. |
+| `1.02.3` | Not strictly compliant | Has a leading `0` in `minor` | :yellow_circle: Fails parsing, but will accept and drop leading `0`'s with a switch. |
+|`1.0.0-beta`| Compliant | Accepted, but `pre-release` definition could be confusing. Suggest using `1.0.0-beta.1` | :green_circle: Parses successfully. |
+| `1.0.0.1` | Not compliant | Uses four parts.  Last delimiter should be `-` or `+` to denote `pre-release` or `build-metadata` | :red_circle: Parsing fails. |
+
+See [`tests`](https://github.com/toitware/toit-semver/tree/main/tests) folder and [`parse-test.toit`](https://github.com/toitware/toit-semver/tree/main/tests/parse-test.toit) for other cases and expected outcomes.
+
+
+## Logical Operators
+The library implements code to support the normal logic operators/comparators,
+such as `>`,  `<=`, etc. The [standard](https://semver.org/) dictates rules
+about these.  Not all are obvious at first.  They operate in the following way:
+| Example | Explanation |
+| - | - |
+| `1.2.3` > `1.2.1` | As expected. |
+| `1.2.3` > `1.2.3-beta` | Versions with pre-release information have a lower precedence than the same without pre-release information. [link](https://semver.org/#spec-item-9).
+| `1.2.3-beta.01` > `1.2.3-beta` | Pre-release information is a set/array, parsed by `.`.  A larger set of pre-release fields has a higher precedence than a smaller set, if all of the preceding identifiers are equal. |
+| `1.2.3-beta.2.1` > `1.2.3-beta.1` | If all of the preceding identifiers are equal, integers must be compared the normal way. |
```

Comment:
Change the higher ones to have a "10" instead of "2"? That makes it clear that it's not just a dump character-by-character comparison.